evm/deploy-contract.js

```diff
@@ -330,12 +389,22 @@ async function processCommand(config, chain, options) {
     printInfo(`${chain.name} | ${contractName}`, contractConfig.address);
 
     await checkContract(contractName, contract, contractConfig);
+
+    return contract;
 }
 
+/**
+ * Main entry point for the deploy-contract script.
+ * Processes deployment options and executes the deployment across specified chains.
+ *
+ * @param {Object} options - Command line options and configuration
+ * @returns {Promise<void>}
+ */
 async function main(options) {
-    await mainProcessor(options, processCommand);
+    await mainProcessor(options, deployEvmContract);
```

Comment:
`processCommand` is just our generic naming scheme for any script, let's keep that?